app/src/main/kotlin/co/netguru/baby/monitor/client/feature/babynotification/NotifyBabyEventUseCase.kt

```diff
@@ -0,0 +1,60 @@
+package co.netguru.baby.monitor.client.feature.babynotification
+
+import co.netguru.baby.monitor.client.feature.firebasenotification.FirebaseNotificationSender
+import co.netguru.baby.monitor.client.feature.firebasenotification.NotificationType
+import dagger.Reusable
+import io.reactivex.Completable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import timber.log.Timber
+import java.util.concurrent.TimeUnit
+
+@Reusable
+class NotifyBabyEventUseCase(
+    private val notificationSender: FirebaseNotificationSender,
+    private val notificationTexts: Map<String, String>
+) {
+    private val babyCryingEvents: PublishSubject<BabyEvent> = PublishSubject.create()
+
+    private fun fetchClientsAndPostNotification(babyEvent: BabyEvent): Completable {
+        val (notificationTitle, notificationType) = when (babyEvent) {
+            BabyEvent.BabyCrying -> notificationTexts[CRY_TITLE_KEY] to NotificationType.CRY_NOTIFICATION
+            BabyEvent.NoiseDetected -> notificationTexts[NOISE_TITLE_KEY] to NotificationType.NOISE_NOTIFICATION
+        }
+        return notificationSender.broadcastNotificationToFcm(
+            notificationTitle ?: error("Notification title missing"),
+            notificationTexts[NOTIFICATION_TEXT_KEY] ?: error("Notification text missing"),
+            notificationType
+        )
+    }
+
+    init {
+        babyCryingEvents
```

Comment:
This will leak as the subscription can live longer that the object with NotifyBabyEventUseCase reference.
I seen that this is used in scope of the service which maybe have been the cause for this - to overcome that:
- expose observable and handle disposable in service 
- create service facade - similliar to classic MVP 
<details>

```kotlin
abstract class BaseServiceWithFacade<T : ServiceFacade, out C : ServiceController<T>> : DaggerService() {

    private lateinit var serviceController: C

    override fun onCreate() {
        super.onCreate()
        attachServiceToController()
    }

    override fun onDestroy() {
        super.onDestroy()
        serviceController.detachService()
    }

    @Suppress("UNCHECKED_CAST")
    private fun attachServiceToController() {
        serviceController = retrieveController()
        serviceController.attachService(this as T)
    }

    protected abstract fun retrieveController(): C
}

interface ServiceFacade

interface ServiceController<T : ServiceFacade> {

    fun attachService(service: T)

    fun detachService()

    fun getService(): T?

}
```
</details>

Reply:
just to be clear -  a/b are both fine and pretty testable i would go with b only if you have some logic in the service currently which is worth of testing